### PR TITLE
Migrate CommentRegularExpressionCheck to Strongly typed AST

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java [hunk 1]
```diff
@@ -19,24 +19,23 @@
...
-  public static final String CHECK_KEY = "CommentRegularExpression";
   private static final String DEFAULT_REGULAR_EXPRESSION = "";
   private static final String DEFAULT_MESSAGE = "The regular expression matches this comment";
 
   private Pattern pattern = null;
 
-  @RuleProperty(
-    key = "regularExpression",
-    defaultValue = "" + DEFAULT_REGULAR_EXPRESSION)
+  @RuleProperty(key = "regularExpression")
```
Comment (andrea-guarino-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java]: Why did you remove the `defaultValue`?

Comment (guillaume-dequenne-sonarsource) [reply python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java]: It was redundant, as DEFAULT_REGULAR_EXPRESSION is already set to ""
